File: ueueo-framework/ueueo-uow/src/main/java/com/ueueo/uow/UnitOfWorkExtensions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using JetBrains.Annotations;

namespace Volo.Abp.Uow;

public static class UnitOfWorkExtensions
{
    public static boolean IsReservedFor(@NonNull this IUnitOfWork unitOfWork, String reservationName)
    {
        Objects.requireNonNull(unitOfWork, nameof(unitOfWork));

        return unitOfWork.IsReserved && unitOfWork.ReservationName == reservationName;
    }

    public static void AddItem<TValue>(@NonNull this IUnitOfWork unitOfWork, String key, TValue value)
        where TValue : class
    {
        Objects.requireNonNull(unitOfWork, nameof(unitOfWork));

        if (!unitOfWork.Items.ContainsKey(key))
        {
            unitOfWork.Items[key] = value;
        }
        else
        {
            unitOfWork.Items.Add(key, value);
        }
    }

    public static TValue GetItemOrDefault<TValue>(@NonNull this IUnitOfWork unitOfWork, String key)
        where TValue : class
    {
        Objects.requireNonNull(unitOfWork, nameof(unitOfWork));

        return unitOfWork.Items.FirstOrDefault(x => x.Key == key).Value.As<TValue>();
    }

    public static TValue GetOrAddItem<TValue>(@NonNull this IUnitOfWork unitOfWork, String key, Func<String, TValue> factory)
        where TValue : class
    {
        Objects.requireNonNull(unitOfWork, nameof(unitOfWork));

        return unitOfWork.Items.GetOrAdd(key, factory).As<TValue>();
    }

    public static void RemoveItem(@NonNull this IUnitOfWork unitOfWork, String key)
    {
        Objects.requireNonNull(unitOfWork, nameof(unitOfWork));

        unitOfWork.Items.RemoveAll(x => x.Key == key);
    }
}

File: ueueo-framework/ueueo-uow/src/main/java/com/ueueo/uow/UnitOfWorkExtensions.java
package com.ueueo.uow;

import org.springframework.lang.NonNull;

import java.util.Objects;
import java.util.function.Function;

public class UnitOfWorkExtensions {
    public static boolean isReservedFor(@NonNull IUnitOfWork unitOfWork, String reservationName) {
        Objects.requireNonNull(unitOfWork);

        return unitOfWork.isReserved() && unitOfWork.getReservationName().equals(reservationName);
    }

    public static <TValue> void addItem(@NonNull IUnitOfWork unitOfWork, String key, TValue value) {
        Objects.requireNonNull(unitOfWork);

        if (!unitOfWork.getItems().containsKey(key)) {
            unitOfWork.getItems().put(key, value);
        } else {
            unitOfWork.getItems().put(key, value);
        }
    }

    public static <TValue> TValue getItemOrDefault(@NonNull IUnitOfWork unitOfWork, String key) {
        Objects.requireNonNull(unitOfWork);
        return (TValue) unitOfWork.getItems().get(key);
    }

    public static <TValue> TValue getOrAddItem(@NonNull IUnitOfWork unitOfWork, String key, Function<String, TValue> factory) {
        Objects.requireNonNull(unitOfWork);
        return (TValue) unitOfWork.getItems().computeIfAbsent(key, factory);
    }

    public static void removeItem(@NonNull IUnitOfWork unitOfWork, String key) {
        Objects.requireNonNull(unitOfWork);

        unitOfWork.getItems().remove(key);
    }
}
